### PR TITLE
WIP: Terraform providers submodule

### PR DESCRIPTION
**Pull request by r4f4**
Use terraform providers from their own repo as a submodule.

### CODE REVIEW

**Comment by patrickdillon** [comment]
Don't tell @zaneb

**Comment by r4f4** [comment]
The onboarding of the `terraform-providers` repo will happen over at: https://github.com/openshift/release/pull/44608

**Comment by zaneb** [comment]
Leaving the merits of submodules aside :upside_down_face: does this mean that there would be no way to test a change to a terraform provider in CI without first merging the change in the provider repo and then proposing a separate PR to the installer that can tell if it worked?

Relatedly, how do we trigger CI jobs for particular platforms when they change in this scenario? Do we end up running every platform's e2e tests any time any one of the Terraform providers change?

**Comment by r4f4** [comment]
> Leaving the merits of submodules aside 🙃 does this mean that there would be no way to test a change to a terraform provider in CI without first merging the change in the provider repo and then proposing a separate PR to the installer that can tell if it worked?

I can't find a way around it. To build the providers separately, they need to be in a separate repo. And that means we need to merge changes there first. If you know of a better way, I'm all ears.
 
> Relatedly, how do we trigger CI jobs for particular platforms when they change in this scenario? Do we end up running every platform's e2e tests any time any one of the Terraform providers change?

I'm trying out `git subtree` in https://github.com/openshift/installer/pull/7621 which retains the whole provider's tree in the Installer repo. Because we can even replay the commits, we can still identify which provider changed and run e2e tests just for the affected platforms.

**Comment by patrickdillon** [comment]
/close
in favor of #7621

File: hack/build.sh [hunk 1]
```diff
@@ -26,6 +26,20 @@ copy_terraform_to_mirror() {
...
+check_terraform_providers() {
+  TARGET_OS_ARCH=$(go env GOOS)_$(go env GOARCH)
+
+  find "${PWD}/terraform/providers/" -maxdepth 1 -mindepth 1 -type d | while read dir; do
+      provider="$(basename "${dir}")"
+	  binDir="${PWD}/terraform/bin/${TARGET_OS_ARCH}"
+      test -f "${binDir}/terraform-provider-${provider}" || return 1
+      test -f "${binDir}/terraform-provider-${provider}.zip" || return 1
+      providerVersion="$(go version -m "${binDir}/terraform-provider-${provider}" | grep -E '^\s+mod' | awk '{ print $2, $3, $4 }')"
+      grep -q -s "$providerVersion" "${dir}/go.sum" || return 1
```
Comment (zaneb) [inline hack/build.sh:39]: should probably add the `-F` flag here I think

**Pull request closed without merging**